### PR TITLE
Add vSphere cloneMode docs

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Description of changes:*
Add docs for vSphere cloneMode feature implemented here https://github.com/aws/eks-anywhere/pull/4634

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/reference/clusterspec/vsphere.md [hunk 1]
```diff
@@ -80,20 +80,21 @@ kind: VSphereMachineConfig
...
+  storagePolicyName: <span style="color:red">"storagePolicy1"</span>  <a href="#storagepolicyname-optional"># Storage policy name associated with VMs</a>
+  template: <span style="color:red">"bottlerocket-kube-v1-25"</span>  <a href="#template-optional"># VM template for EKS Anywhere (required for RHEL/Ubuntu-based OVAs)</a>
+  cloneMode: <span style="color:red">"fullClone"</span>               <a href="#clonemode-optional"># Clone mode to use when cloning VMs from the template</a>
+  users:                               <a href="#users-optional"># Add users to access VMs via SSH</a>
+  - name: <span style="color:red">"ec2-user"</span>                   <a href="#users0name-optional"># Name of each user set to access VMs</a>
+    sshAuthorizedKeys:                 <a href="#users0sshauthorizedkeys-optional"># SSH keys for user needed to access VMs</a>
     - <span style="color:red">"ssh-rsa AAAAB3NzaC1yc2E..."</span>
-  tags:                              <a href="#tags-optional"># List of tags to attach to cluster VMs, in URN format</a>
+  tags:                                <a href="#tags-optional"># List of tags to attach to cluster VMs, in URN format</a>
```
Comment (abhinavmpandey08) [inline docs/content/en/docs/reference/clusterspec/vsphere.md:97]: Most of the changes here are to fix the indentation for the comments. Here's how it looks rendered
image

Comment (chrisnegus) [reply docs/content/en/docs/reference/clusterspec/vsphere.md:97]: Sure. That's fine with me.

File: docs/content/en/docs/reference/clusterspec/vsphere.md [hunk 2]
```diff
@@ -319,6 +320,18 @@ The VM template to use for your EKS Anywhere cluster. This template was created
 [imported the OVA file into vSphere]({{< relref "../vsphere/vsphere-ovas.md" >}}).
 This is a required field if you are using Ubuntu-based or RHEL-based OVAs.
 
+### cloneMode (optional)
+`cloneMode` defines the clone mode to use when creating the cluster VMs from the template . Allowed values are:
```
Comment (chrisnegus) [inline docs/content/en/docs/reference/clusterspec/vsphere.md:324]:
```suggestion
`cloneMode` defines the clone mode to use when creating the cluster VMs from the template. Allowed values are:
```

**Review by chrisnegus — COMMENTED** [review]
Looks good. Just one little thing.

**Pull request merged**